error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/AssertJPrimitiveRules.java

```diff
@@ -19,15 +25,49 @@ static final class AssertThatIsEqualTo {
     @BeforeTemplate
     AbstractBooleanAssert<?> before(boolean actual, boolean expected) {
       return Refaster.anyOf(
-          assertThat(actual == expected).isTrue(), assertThat(actual != expected).isFalse());
+          assertThat(actual).isSameAs(expected), assertThat(actual).isSameAs(expected));
     }
 
     @BeforeTemplate
-    @SuppressWarnings(
-        "java:S1244" /* The (in)equality checks are fragile, but may be seen in the wild. */)
-    AbstractBooleanAssert<?> before(double actual, double expected) {
+    AbstractByteAssert<?> before(byte actual, byte expected) {
+      return Refaster.anyOf(
+          assertThat(actual).isSameAs(expected), assertThat(actual).isSameAs(expected));
+    }
+
+    @BeforeTemplate
+    AbstractCharacterAssert<?> before(char actual, char expected) {
+      return Refaster.anyOf(
+          assertThat(actual).isSameAs(expected), assertThat(actual).isSameAs(expected));
+    }
+
+    @BeforeTemplate
+    AbstractShortAssert<?> before(short actual, short expected) {
+      return Refaster.anyOf(
+          assertThat(actual).isSameAs(expected), assertThat(actual).isSameAs(expected));
+    }
+
+    @BeforeTemplate
+    AbstractIntegerAssert<?> before(int actual, int expected) {
+      return Refaster.anyOf(
+          assertThat(actual).isSameAs(expected), assertThat(actual).isSameAs(expected));
+    }
+
+    @BeforeTemplate
+    AbstractLongAssert<?> before(long actual, long expected) {
+      return Refaster.anyOf(
+          assertThat(actual).isSameAs(expected), assertThat(actual).isSameAs(expected));
+    }
+
+    @BeforeTemplate
+    AbstractFloatAssert<?> before(float actual, float expected) {
+      return Refaster.anyOf(
+          assertThat(actual).isSameAs(expected), assertThat(actual).isSameAs(expected));
+    }
```

Comment:
Here and below: due to overload matching we now need to specify all variants explicitly.